### PR TITLE
Add HW3 tests (smoke/one-shot/edge/pattern)

### PR DESCRIPTION
**Pull request by Muwen320**
This PR adds four HW3 tests for classify_by_name:

- Smoke test: test_smoke_basic_repo_name
- One-shot test: test_one_shot_known_language_mapping
- Edge test: test_edge_numeric_and_very_long
- Pattern test: test_pattern_many_python_repos

Author: Muwen320  
Reviewer: Wenshu0206

All tests pass locally.

### CODE REVIEW

File: hw3-test/test_hw3_muwen320.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    author: Muwen320
+    reviewer: Wenshu0206
+    category: smoke test
+    """
+    name = "my-django-app"
+    result = classify_by_name(name)
+    assert isinstance(result, str)
+    assert result != ""
+
```
Comment (Wenshu0206) [inline hw3-test/test_hw3_muwen320.py:16]: This is a simple and effective smoke test. It checks that classify_by_name runs without errors, returns a string, and does not produce an empty result. The test is easy to understand and appropriate for a smoke-level check.

optional improvements:
If classify_by_name has well-defined output categories, asserting membership in a known set would make the test stronger—but for a smoke test, the current checks are sufficient.

Overall, the test is straightforward, clean, and fits its purpose well.

File: hw3-test/test_hw3_muwen320.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+def test_one_shot_known_language_mapping():
+    """
+    author: Muwen320
+    reviewer: Wenshu0206
+    category: one-shot test
+    """
+    name = "system-design-primer"
+    result = classify_by_name(name)
+    assert result == "Python"
+
```
Comment (Wenshu0206) [inline hw3-test/test_hw3_muwen320.py:27]: This test is clear and focused: it verifies that classify_by_name correctly maps a well-known repository name to the expected language (“Python”). The intent of a one-shot test is satisfied—checking a single known input–output pair.

Hard-coding "Python" is fine if the mapping is stable, but if the model or mapping logic changes over time, this test may become brittle.

Overall, the test is straightforward, readable, and suitable as a one-shot correctness check.

**Review by Wenshu0206 — APPROVED** [review]
